common/lib/capa/capa/capa_problem.py

```diff
@@ -191,6 +193,48 @@ def __init__(self, problem_text, id, capa_system, state=None, seed=None):
 
         self.extracted_tree = self._extract_html(self.tree)
 
+    def make_xml_compatible(self, tree):
+        """
+        Adjust tree xml in-place for compatibility, such as
+        supporting an old xml format by translating it to a new format.
```

Comment:
My idea here was to make a sort of official choke-point that manages xml coming in from storage before it goes to the runtime. I think that's ultimately going to be a very useful feature, where you spell out various "compat" formats, but without that complexity spreading through the whole runtime. Anyway, in reality, there's just two cases handled here, so I'll promote the comments as you say and write some more.